docs/providers/resource.md

```diff
@@ -15,3 +15,89 @@ def create_sync_resource() -> typing.Iterator[str]:
 class DIContainer(BaseContainer):
     sync_resource = providers.Resource(create_sync_resource)
 ```
+
+# These use cases demonstrate how to use `ContextResource` and `AsyncContextResource` within the `DIContainer`.
+## Use Case 1: Synchronous Context Resource
+
+```python
+import datetime
+import logging
+from contextlib import contextmanager
+from that_depends.providers.context_resources import ContextResource
+
+logger = logging.getLogger(__name__)
+
+# Define a synchronous context resource creator
+def create_sync_context_resource() -> typing.Iterator[datetime.datetime]:
+    logger.debug("Resource initiated")
+    yield datetime.datetime.now(tz=datetime.timezone.utc)
+    logger.debug("Resource destructed")
+
+class DIContainer:
+    sync_context_resource = ContextResource(create_sync_context_resource)
+```
+
+## Use Case 2: Asynchronous Context Resource
+
+```python
+import datetime
+import logging
+from contextlib import asynccontextmanager
+import asyncio
+from that_depends.providers.context_resources import AsyncContextResource
+
+logger = logging.getLogger(__name__)
+
+async def create_async_context_resource() -> typing.AsyncIterator[datetime.datetime]:
+    logger.debug("Async resource initiated")
+    yield datetime.datetime.now(tz=datetime.timezone.utc)
+    logger.debug("Async resource destructed")
+
+# Define the DIContainer with the asynchronous context resource
+class DIContainer:
+    async_context_resource = AsyncContextResource(create_async_context_resource)
+
+async def use_async_context_resource():
+    resource = await DIContainer.async_context_resource.async_resolve()
+    print(f"Asynchronous resource value: {resource}")
+
+asyncio.run(use_async_context_resource())
+```
+
+## Use Case 3: Overriding Context Resources
```

Comment:
There is no need to describe overriding for context resources, because it's the same for all providers